model/event/BatteryBeforeRemoveEvent.php

```diff
@@ -19,6 +19,29 @@
 
 namespace oat\taoBattery\model\event;
 
-class BatteryBeforeRemoveEvent extends AbstractBatteryEvent
+class BatteryBeforeRemoveEvent
 {
+    /**
+     * @var string
+     */
+    protected $batteryId;
+
+
+    /**
+     * AbstractBatteryEvent constructor.
+     * @param string $batteryId
+     */
+    public function __construct($batteryId)
+    {
+        $this->batteryId = $batteryId;
+    }
+
+    /**
+     * @return string
+     * @deprecated
+     */
+    public function getBatteryId()
```

Comment:
You can implement it using `$this->battery->getUri()` and keep class like another events removing `$this->batteryId`

Reply:
I can't remove this as it is used in rework assignment branch.

Reply:
My idea is to unify BatteryBeforeRemoveEvent with all other battery events (extend AbstractBatteryEvent):
1. change `new BatteryBeforeRemoveEvent( ... )` call passing battery instead of battery_id
2. add temporary 
```
public function getBatteryId() {
    return $this->battery->getUri();
}
```
to provide compatibility with code in assignment rework branch

Reply:
I don't know what it is used for. Because of that, I don't want to modify this.

Reply:
Ok, could you simply unify `BatteryBeforeRemoveEvent` as well as other events (remove `getBatteryId` method)? it will looks ok for all code in develop branches and will not require us to return and modify taoBattery again. After this PR merged I will adapt taoNccerDelivery assignment rework branch

Reply:
Should it just implement BatteryEventInterface or should it extend AbstractBatteryEvent?